internal/core/data/v2/controller/http/core/data/event.go

```diff
@@ -0,0 +1,76 @@
+package data
+
+import (
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"net/http"
+
+	v2container "github.com/edgexfoundry/edgex-go/internal/core/data/v2/bootstrap/container"
+	"github.com/edgexfoundry/edgex-go/internal/core/data/v2/handler/common"
+	handler "github.com/edgexfoundry/edgex-go/internal/core/data/v2/handler/core/data"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/errorconcept"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	commonDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	requestDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
+	respDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/responses"
+)
+
+func EventController(w http.ResponseWriter, r *http.Request, dic *di.Container) {
```

Comment:
@lenny-intel Lenny, as we proposed the v2 API design architecture, we separate the code structure into 3 layers:

- Transport (UI) Layer: The controller functions. Transport-specific logic implementation, such as REST(HTTP) or MQTT.
- Handler (Application) Layer: The handler functions. Business logic or use-case implementation.
- Infrastructure (Persistence) Layer: The infrastructure functions. Repository contract and implementation for different persistences.

The `EventController` function handles the HTTP request, processes the content as input to the controller function, therefore the naming here is `Controller`.
If any other suggestion or concern about this, please let us know, thanks.

Reply:
My point is the  standard `Handler` naming convention, as you have used in the description above. `Controller` to me and is suspect others has meaning that has more responsibility that a simple `Handler`. When we see the name with `Handler` in it we know right away it is for a specific REST endpoint.

Reply:
Currently, in `Handler` signature, there is no protocol specific argument, such as httpRequest or httpResponse.
It will be a general business logic layer for the future service interaction protocols (different Controllers).
If we rename `Handler` to `Processor`, is it ok?